packages/vite/src/node/plugins/resolve.ts

```diff
@@ -375,12 +376,12 @@ export function tryNodeResolve(
     path.isAbsolute(importer) &&
     fs.existsSync(cleanUrl(importer))
   ) {
-    basedir = path.dirname(importer)
+    basedir = fs.realpathSync.native(path.dirname(importer))
   } else {
     basedir = root
   }
 
```

Comment:
To further elaborate Ben's comment about `nestedResolveFrom`, which would be on this line in the `main` branch. It's currently used to resolve `optimiseDeps.include` with syntax like `my-lib > some-other-lib`. It's probably fine to not update `nestedResolveFrom` as it's currently used to resolve CJS dependencies only, but yeah ideally we want to move away from it too.

Reply:
Thanks for clarifying. The change to `nestedResolveFrom` can be done in a future PR 👍